summingbird-scalding/src/main/scala/com/twitter/summingbird/scalding/store/DirectoryBatchedStore.scala

```diff
@@ -81,7 +81,7 @@ class DirectoryBatchedStore[K <: Writable, V <: Writable](val rootPath: String)(
 
         val lastBatchStatus =
           hdfsPaths.map(getFileStatus(_, conf))
-            .filter { input => input._1 && (BatchID(input._2) < exclusiveUB) }
+            .filter { input => input._1 && (BatchID(input._2.milliSinceEpoch) < exclusiveUB) }
```

Comment:
by the way, that `apply(ts: Timestamp)` just seems like a broken method. How did that stick around so long?